lib/iCal.php

```diff
@@ -195,56 +194,38 @@ public static function generate_ical_feed( $post = null ) {
 
 		foreach ( $events_posts as $event_post ) {
 			// add fields to iCal output
-			$item   = array();
+			$item = array();
 
-			$startDate = date( 'Ymd\THis', strtotime( $event_post->EventStartDate ) );
-			$endDate   = date( 'Ymd\THis', strtotime( $event_post->EventEndDate ) );
-
-			$tz_startDate = self::wp_date( 'Ymd\THis\Z', strtotime( $event_post->EventStartDate ) );
-			$tz_endDate = self::wp_date( 'Ymd\THis\Z', strtotime( $event_post->EventEndDate ) );
+			$full_format = 'Ymd\THis';
+			$time = (object) array(
+				'start' => self::wp_strtotime( $event_post->EventStartDate ),
+				'end' => self::wp_strtotime( $event_post->EventEndDate ),
+				'modified' => self::wp_strtotime( $event_post->post_modified ),
+				'created' => self::wp_strtotime( $event_post->post_date ),
+			);
 
 			if ( 'yes' == get_post_meta( $event_post->ID, '_EventAllDay', true ) ) {
-				$startDate = substr( $startDate, 0, 8 );
-				$endDate   = substr( $endDate, 0, 8 );
-				// endDate bumped ahead one day to counter iCal's off-by-one error
-				$endDateStamp = strtotime( $endDate );
-				$endDate      = date( 'Ymd', $endDateStamp + 86400 );
-				$type         = 'DATE';
-				$item[] = "DTSTART;VALUE=$type:" . $startDate;
-				$item[] = "DTEND;VALUE=$type:" . $endDate;
-
+				$type = 'DATE';
+				$format = 'Ymd';
 			} else {
 				$type = 'DATE-TIME';
+				$format = 'Ymd\THis\Z';
+			}
 
-				if ( ! empty( $wp_timezone ) ){
-					$item[] = "DTSTART;TZID=\"{$wp_timezone}\":{$tz_startDate}";
-				}
-				$item[] = 'DTSTART:' . $tz_startDate;
-				$item[] = 'DTSTART:' . $startDate;
+			$tzoned = (object) array(
+				'start' => date( $format, $time->start ),
+				'end' => date( $format, $time->end ),
```

Comment:
It looks like it is using local time and claiming it is UTC.  So January 24, 2015 03:00 EST would be translated to January 24, 2015 03:00 UTC.